src/callbacks_step/amr_dg2d.jl

```diff
@@ -313,7 +313,7 @@ function coarsen!(u_ode::AbstractVector, adaptor,
         reinitialize_containers!(mesh, equations, dg, cache)
 
         resize!(u_ode,
-                nvariables(equations) * nnodes(dg)^ndims(mesh) * nelements(dg, cache))
+                nvariables(equations) * nnodes(dg)^2 * nelements(dg, cache))
         u = wrap_array(u_ode, mesh, equations, dg, cache)
```

Comment:
I think parts like these become much harder to understand with this PR. What do you think, @sloede?

Reply:
Yes, I have to agree with Hendrik.

The advantage of explicitly adding `ndims(mesh)` is that it makes the factor self-explanatory. Also in the other parts of the code, one avoids having to explain why a factor is what it is and how it was computed, or even how to change it when switching to another dimension.

I see the argument for more "elegant" code (avoiding the impression of having a varying dimension). However, I've seen too many codes where "magic" constants appear out of nowhere, and everyone reading it after 3 years has to re-understand what the intent/thought process of the original author was.

IMHO, the current version is concise and self-explanatory in terms of what is being done, thus I'd prefer to keep it as it is.

Reply:
To throw in my two cents. I want to keep the `ndims` calls in the code. I think it is much more informative to have terms like `2^ndims(mesh) - 1` rather than values like `3` and `7`. This is coming from the perspective of someone who recently became very familiar with the AMR implementation as I debugged and tracked down the loss of conservation. I think the current code is perfectly understandable.

Reply:
Ok, I will close this PR then. Opinions about the issue?

Reply:
I would close it in light of the comments above